src/interp.rs

```diff
@@ -188,12 +224,14 @@ impl<'a, 'tcx> Machine<'a, 'tcx> {
             Rvalue::Aggregate(ref kind, ref ops) => {
                 match **kind {
                     AggregateKind::Tuple => {
-                        // XXX: Tuple layout needs some proper thought, and
-                        // should be upcoming in the next PR, so we handle the
-                        // empty tuple explicitly as this is the bottom type in
-                        // Rust.
-                        if ops.len() != 0 {
-                            unimplemented!()
+                        if ops.len() == 0 {
+                            return; // Bottom type is empty tuple, and zero-sized.
```

Comment:
Is "bottom" here in the sense of a lattice? The Rust terminology is `unit` for `()`.

Reply:
Yeah I should say `unit` type. Fixed